drivers/video/gc2145.c

```diff
@@ -768,14 +768,22 @@ struct gc2145_ctrls {
 struct gc2145_data {
 	struct gc2145_ctrls ctrls;
 	struct video_format fmt;
+	struct video_rect crop;
+	uint16_t format_width;
+	uint16_t format_height;
+	uint8_t ratio;
 };
 
-#define GC2145_VIDEO_FORMAT_CAP(width, height, format)                                             \
-	{                                                                                          \
-		.pixelformat = format, .width_min = width, .width_max = width,                     \
-		.height_min = height, .height_max = height, .width_step = 0, .height_step = 0,     \
+#define GC2145_VIDEO_FORMAT_CAP_HL(width_l, width_h, height_l, height_h, format)          \
+	{                                                                                     \
+		.pixelformat = format, .width_min = width_l, .width_max = width_h,                \
+		.height_min = height_l, .height_max = height_h, .width_step = 0, .height_step = 0,\
 	}
 
+#define GC2145_VIDEO_FORMAT_CAP(width, height, format)	\
+	GC2145_VIDEO_FORMAT_CAP_HL(width, width, height, height, format)
```

Comment:
By the way, could you add parentheses to arguments where used in the macro value:
```suggestion
#define GC2145_VIDEO_FORMAT_CAP_HL(width_l, width_h, height_l, height_h, format)          \
	{                                                                                          \
		.pixelformat = (format), .width_min = (width_l), .width_max = (width_h),                \
		.height_min = (height_l), .height_max = (height_h), .width_step = 0, .height_step = 0,\
	}

#define GC2145_VIDEO_FORMAT_CAP(width, height, format)	\
	GC2145_VIDEO_FORMAT_CAP_HL((width), (width), (height), (height), (format))
```